### PR TITLE
Make the navigation wrapping ul component user customisable

### PR DESCRIPTION
**Pull request by mikehazell**
The `NavigationContainer` component rendered it's children inside a `ul` meaning if you wanted to render anything other than an `li` you would have to do some gymnastics to make it work.

This change makes it the users' responsibility to properly wrap list items in a `ul`.
- exposes a `NavItemGroup` component with appropriate styling
- updates "Custom Admin UI Navigation" usage docs

### CODE REVIEW

File: .changeset/real-numbers-bake.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+"@keystone-6/core": major
```
Comment (mikehazell) [inline .changeset/real-numbers-bake.md:2]: Technically a breaking change? ... custom Navigation in the Admin UI will render weirdly or have invalid DOM structure if not updated.

File: docs/pages/docs/guides/custom-admin-ui-navigation.md [hunk 2]
```diff
@@ -292,13 +307,50 @@ Otherwise, all lists will be added.
...
+          {group.lists.map((key) => {
+            const list = lists.find((l) => l.key === key);
+            return list ? <ListNavItem key={key} list={list} /> : null;
+          })}
+        </NavItemGroup>
+      </Box>
+    ))}
+  </NavigationContainer>
+)
+```
```
Comment (mikehazell) [inline docs/pages/docs/guides/custom-admin-ui-navigation.md:352]: Kind of contrived, you could do this with the `ListNavItems` component too but I wanted to show that this component is also exported.

File: docs/pages/docs/guides/custom-admin-ui-navigation.md [hunk 3]
```diff
@@ -323,6 +375,29 @@ type NavItemProps = {
...
+### NavItemGroup
+
+This component is a styled unordered list `<ul>` which should be used to wrap `NavItem`, `ListNavItem` and `comp
+
+```tsx
+import { NavItemGroup } from '@keystone-6/core/admin-ui/components'
+```
+
+{% hint kind="warn" %}
+Versions of `@keystone-6/core` before `1.2.0` wrapped all children of `NavigationContainer` with the `NavItemGroup` component.
```
Comment (mikehazell) [inline docs/pages/docs/guides/custom-admin-ui-navigation.md:387]: I've guessed what the version number will be. Also I'm not sure we need to call out that this thing has changed.

**Comment by dcousens** [comment]
Likely superseded by https://github.com/keystonejs/keystone/pull/9253

**Comment by dcousens** [comment]
Superseded by #9253, and if not, we'll re-open

**Pull request closed without merging**